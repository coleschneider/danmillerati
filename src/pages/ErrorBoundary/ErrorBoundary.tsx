import * as React from "react";

interface State {
    error: Error | null;
}

interface Props {
    children: React.ReactElement | React.ReactFragment[];
}

class ErrorBoundary extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            error: null
        };
    }

    componentDidCatch(error: Error) {
        this.setState({
            error
        });
    }

    render() {
        const { error } = this.state;
        const { children } = this.props;
        if (error) {
            return (
                <div className="section-error">
                    <h2 test-id="text-error">Oh-no! Something went wrong</h2>
                </div>
            );
        }
        return children;
    }
}

export default ErrorBoundary;
