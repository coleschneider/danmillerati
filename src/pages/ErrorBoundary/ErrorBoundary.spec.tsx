import * as React from "react";
import { ShallowWrapper } from "enzyme";
import ErrorBoundary from "./ErrorBoundary";

const BuggyComponent = () => null;

describe("ErrorBoundary", () => {
    let wrapper: ShallowWrapper;
    const mountErrorBoundary = () => {
        wrapper = mount(
            <ErrorBoundary>
                <BuggyComponent />
            </ErrorBoundary>
        );
    };
    it("should match the snapshot", () => {
        mountErrorBoundary();
        expect(wrapper).toMatchSnapshot();
    });
    it("should display an error if the child component throws", () => {
        mountErrorBoundary();
        wrapper.find(BuggyComponent).simulateError(new Error("test error"));
        expect(wrapper.find('[test-id="text-error"]').text()).toBe(
            "Oh-no! Something went wrong"
        );
    });
});
