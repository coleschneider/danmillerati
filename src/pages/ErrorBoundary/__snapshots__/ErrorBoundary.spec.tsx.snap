// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should match the snapshot 1`] = `
<ErrorBoundary>
  <BuggyComponent />
</ErrorBoundary>
`;
